consul/data_source_consul_peering.go

```diff
@@ -0,0 +1,103 @@
+package consul
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceConsulPeering() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceConsulPeeringRead,
+		Schema: map[string]*schema.Schema{
+			// In
+			"peer_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"datacenter": {
```

Comment:
We should probably get rid of this parameter. Consul will not forward the HTTP/RPC requests to read a peering to other datacenters so this would have no effect.